cumulus_library/studies/core/core_templates/allergyintolerance.sql.jinja

```diff
@@ -0,0 +1,138 @@
+    SELECT
+         {{-
+            utils.basic_cols(
+                'allergyintolerance',
+                'a',
+                [
+                    'id',
+                ]
+            )
+        }},
+        {{-
+            utils.nullable_cols(
+                'allergyintolerance',
+                'a',
+                [
+                    'type',
+                    'category',
+                    'criticality',
+                    ('patient', 'reference', 'patient_ref'),
+                    ('encounter', 'reference', 'encounter_ref'),
+                ],
+                schema
+            )
+        }},
+        {{-
+            utils.date_cols_from_str(
+                'allergyintolerance',
+                'a',
+                ['recordedDate'],
+                schema
+            )
+        }},
+        {{-
+            utils.truncate_date_cols(
+                'allergyintolerance',
+                'a',
+                [
+                    ('recordedDate', 'week'),
+                    ('recordedDate', 'month'),
+                    ('recordedDate', 'year'),
+                ],
+                schema
+            )
+        }}
+    FROM allergyintolerance AS a
+),
+
+temp_category AS (
+    SELECT
+        a.id,
+        t.category
+    FROM
+        allergyintolerance AS a,
+        UNNEST(a.category) AS t (category)
+),
+
+flattened_reaction AS ({{ unnest_utils.flatten('allergyintolerance', 'reaction') }}),
+
+temp_reaction AS (
+    SELECT
+        r.id,
+        r.row,
+        dn_subs.code AS substance_code,
+        dn_subs.system AS substance_system,
+        dn_subs.display AS substance_display,
+        dn_man.code AS manifestation_code,
+        dn_man.system AS manifestation_system,
+        dn_man.display AS manifestation_display,
+        {{-
+            utils.nullable_cols(
+                'allergyintolerance',
+                'r',
+                [
+                    ('reaction', 'severity', 'severity'),
+                ],
+                schema
+            )
+        }}
+    FROM flattened_reaction AS r
+    LEFT JOIN core__allergyintolerance_dn_reaction_substance AS dn_subs
+        ON r.id = dn_subs.id AND r.row = dn_subs.row
+    LEFT JOIN core__allergyintolerance_dn_reaction_manifestation AS dn_man
+        ON r.id = dn_man.id AND r.row = dn_man.row
+)
+
+SELECT
+    ta.id,
+    CONCAT('AllergyIntolerance/', ta.id) AS allergyintolerance_ref,
+
+    {#- We don't expose system for these next two since we filter
+        down to a single system in the denormalization table #}
+    dn_cstat.code AS clinicalStatus_code,
+    dn_vstat.code AS verificationStatus_code,
+
+    {#- type, category, and criticality are not in US Core.
+        But they are useful for clinical interpretation. #}
+    ta.type,
+    tcat.category,
+    ta.criticality,
+
+    dn_code.code AS code_code,
+    dn_code.system AS code_system,
+    dn_code.display AS code_display,
+
+    ta.patient_ref,
```

Comment:
This is the first core resource with a patient field instead of subject - I'm assuming our vibes here are to keep consistent with the spec? But this patient/subject difference is honestly a headache. In the count code, I chose to keep the output field as cnt_subject_ref regardless.